packages/contracts/src/contracts/current/test/DummyERC721Receiver/DummyERC721Receiver.sol

```diff
@@ -0,0 +1,63 @@
+/*
+The MIT License (MIT)
+
+Copyright (c) 2016 Smart Contract Solutions, Inc.
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be included
+in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+pragma solidity ^0.4.24;
+
+import "../../tokens/ERC721Token/IERC721Receiver.sol";
+
+contract DummyERC721Receiver is
+    IERC721Receiver
+{
+
+    event TokenReceived(
+        address from,
+        uint256 tokenId,
+        bytes data
+    );
+
+    /**
+    * @notice Handle the receipt of an NFT
+    * @dev The ERC721 smart contract calls this function on the recipient
+    *  after a `safetransfer`. This function MAY throw to revert and reject the
+    *  transfer. This function MUST use 50,000 gas or less. Return of other
+    *  than the magic value MUST result in the transaction being reverted.
+    *  Note: the contract address is always the message sender.
+    * @param _from The sending address
+    * @param _tokenId The NFT identifier which is being transfered
+    * @param _data Additional data with no specified format
+    * @return `bytes4(keccak256("onERC721Received(address,uint256,bytes)"))`
+    */
+    function onERC721Received(
+        address _from,
+        uint256 _tokenId,
+        bytes _data
+    )
+        public
+        returns (bytes4)
+    {
+        emit TokenReceived(_from, _tokenId, _data);
+        return ERC721_RECEIVED;
```

Comment:
Where is `ERC721_RECEIVED` declared?

Reply:
In the ERC721 Receiver interface -- `tokens/ERC721Token/IERC721Receiver.sol`